src/addresstype.cpp

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <addresstype.h>
+
+static const std::string ADDRESS_TYPE_STRING_BASE58 = "legacy";
+static const std::string ADDRESS_TYPE_STRING_BECH32 = "bech32";
+
+Optional<AddressType> ParseAddressType(const std::string& type)
+{
+    if (type == ADDRESS_TYPE_STRING_BASE58) {
+        return AddressType::BASE58;
+    } else if (type == ADDRESS_TYPE_STRING_BECH32) {
+        return AddressType::BECH32;
+    }
+    return nullopt;
+}
+
+const std::string& FormatAddressType(AddressType type)
+{
+    switch (type) {
+    case AddressType::BASE58: return ADDRESS_TYPE_STRING_BASE58;
+    case AddressType::BECH32: return ADDRESS_TYPE_STRING_BECH32;
+    default: assert(false);
```

Comment:
Could remove the default case and move the assert after the switch, so that the compiler will issue a warning when the assert could be hit

Reply:
The compiler throws plenty of warnings even if I remove the `assert`:
<img width="749" alt="Schermafbeelding 2019-03-14 om 10 54 05" src="https://user-images.githubusercontent.com/10217/54347597-95b80c00-4647-11e9-89fa-6aad4c77a4e5.png">